synfig-studio/src/gui/workarea.cpp

```diff
@@ -767,6 +768,14 @@ void WorkArea::set_onion_skins(int *onions)
 	save_meta_data();
 }
 
+void
+WorkArea::set_onion_skin_keyframes(bool x)
+{
+	if (onion_skin_keyframes == x)
+		return;
+	onion_skin_keyframes = x;
```

Comment:
You have to add here `save_meta_data()` and update this method as well. Otherwise when user reopens this synfig document, it will always be set as Onion Skin per frame, not keyframe.

Reply:
You have to call `queue_draw()`, don't you?

Reply:
Yup :+1: